api/server/sdk/volume_ops.go

```diff
@@ -652,199 +659,223 @@ func setSpecBool(current, req bool, reqSet interface{}) bool {
 	return current
 }
 
-// GetEnforcedVolSpecs returns volume spec merged with enforced policy applied if any
-func GetEnforcedVolSpecs(locator *api.VolumeLocator, spec *api.VolumeSpec) (*api.VolumeSpec, error) {
-	if locator != nil {
-		// if volume label is provided, override default policy
-		if val, ok := locator.VolumeLabels["enforce_disable"]; ok {
-			if val == "true" {
-				return spec, nil
-			}
-		}
-	}
+// GetDefaultVolSpecs returns volume spec merged with default storage policy applied if any
+func GetDefaultVolSpecs(
+	ctx context.Context,
+	spec *api.VolumeSpec,
+	isUpdate bool,
+) (*api.VolumeSpec, error) {
 	storPolicy, err := policy.Inst()
 	if err != nil {
 		return nil, status.Errorf(codes.Internal, "Unable to get storage policy instance %v", err)
 	}
 
+	var policy *api.SdkStoragePolicy
 	// check if custom policy passed with volume
 	if spec.GetStoragePolicy() != "" {
 		inspReq := &api.SdkOpenStoragePolicyInspectRequest{
+			// name of storage policy specified in volSpecs
 			Name: spec.GetStoragePolicy(),
 		}
-		customPolicy, customErr := storPolicy.Inspect(context.Background(), inspReq)
+		// inspect will make sure user will atleast have read access
+		customPolicy, customErr := storPolicy.Inspect(ctx, inspReq)
 		if customErr != nil {
 			return nil, customErr
 		}
-		// update spec with custom storage policy
-		updatedSpec := mergeVolumeSpecsPolicy(spec, customPolicy.GetStoragePolicy().GetPolicy())
-		return updatedSpec, nil
+
+		// should we only allow policy to be used by owner who has write/admin access
+		if !customPolicy.GetStoragePolicy().IsPermitted(ctx, api.Ownership_Admin) {
+			return nil, status.Errorf(codes.PermissionDenied, "Cannot use storage policy %v", customPolicy.GetStoragePolicy().GetName())
+		}
+		policy = customPolicy.GetStoragePolicy()
+	} else {
+		// check if default storage policy is set
+		defPolicy, err := storPolicy.DefaultInspect(context.Background(), &api.SdkOpenStoragePolicyDefaultInspectRequest{})
+		if err == kvdb.ErrNotFound || defPolicy.GetStoragePolicy() == nil {
+			// no default storage policy,
+			// should we add NA as volume is not created by any storage policy here ?
+			return spec, nil
+		}
+		// err means there is policy stored, but we are not able to retrive it
+		// hence we are not allowing volume create operation
+		if err != nil {
```

Comment:
use `else if` to make it clear that this is part of the previous `if` statement.